src/lir/converter/instruction.rs

```diff
@@ -0,0 +1,375 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use solang_parser::pt::Loc;
+
+use crate::codegen::cfg::Instr;
+use crate::lir::converter::Converter;
+use crate::lir::expressions::Operand;
+use crate::lir::instructions::Instruction;
+use crate::lir::vartable::Vartable;
+
+impl Converter<'_> {
+    pub(crate) fn lower_instr(
+        &self,
+        instr: &Instr,
+        vartable: &mut Vartable,
+        results: &mut Vec<Instruction>,
+    ) {
+        match instr {
+            Instr::Nop => {
+                results.push(Instruction::Nop);
+            }
+            Instr::Set { res, expr, loc, .. } => {
+                // [t] a = b + c * d
+                // converts to:
+                //   1. [t1] tmp_1 = c * d;
+                //   2. [t2] tmp_2 = b + tmp_1
+                //   3. [t] a = tmp_2;
+                let dest_operand = vartable.get_operand(res, *loc);
+                self.lower_expression(&dest_operand, expr, vartable, results);
+            }
+            Instr::Store { dest, data } => {
+                // type checking the dest.ty() and data.ty()
+                let dest_op = self.to_operand_and_insns(dest, vartable, results);
+                let data_op = self.to_operand_and_insns(data, vartable, results);
+                results.push(Instruction::Store {
+                    loc: /*missing from cfg*/ Loc::Codegen,
```

Comment:
Should this be fixed in codegen? Or rather be `Loc::Lir`?

Reply:
Hi @xermicus, thank you for your time reviewing my PR.
Places commented with `/*missing from cfg*/` should be fixed in the codegen.